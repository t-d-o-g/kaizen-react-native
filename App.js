/* eslint no-underscore-dangle: 0 */

import React from 'react'
import { Platform, StatusBar, StyleSheet, View } from 'react-native'
import { AppLoading, Asset, Font, Icon } from 'expo'
import PropTypes from 'prop-types'
import AppNavigator from './src/navigation/AppNavigator'
const robotDev = require('./assets/images/robot-dev.png')
const robotProd = require('./assets/images/robot-prod.png')
const spaceMonoRegular = require('./assets/fonts/SpaceMono-Regular.ttf')

export default class App extends React.Component {
  state = {
    isLoadingComplete: false,
  }

  static propTypes = {
    skipLoadingScreen: PropTypes.bool,
  }

  _loadResourcesAsync = async () =>
    Promise.all([
      Asset.loadAsync([robotDev, robotProd]),
      Font.loadAsync({
        // This is the font that we are using for our tab bar
        ...Icon.Ionicons.font,
        // We include SpaceMono because we use it in HomeScreen.js. Feel free
        // to remove this if you are not using it in your app
        'space-mono': spaceMonoRegular,
      }),
    ])

  _handleLoadingError = error => {
    // In this case, you might want to report the error to your error
    // reporting service, for example Sentry
    // eslint-disable-next-line no-console
    console.warn(error)
  }

  _handleFinishLoading = () => {
    this.setState({ isLoadingComplete: true })
  }

  render() {
    App.defaultProps = {
      skipLoadingScreen: undefined,
    }
    const { isLoadingComplete } = this.state
    const { skipLoadingScreen } = this.props
    if (!isLoadingComplete && !skipLoadingScreen) {
      return (
        <AppLoading
          startAsync={this._loadResourcesAsync}
          onError={this._handleLoadingError}
          onFinish={this._handleFinishLoading}
        />
      )
    }
    return (
      <View style={styles.container}>
        {Platform.OS === 'ios' && <StatusBar barStyle="default" />}
        <AppNavigator />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
})
