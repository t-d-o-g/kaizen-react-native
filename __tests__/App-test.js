/* eslint-disable */

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import NavigationTestUtils from 'react-navigation/NavigationTestUtils'
import App from '../App'

/* Workaround for TypeError: Cannot read property 'setNativeProps' of null
   https://github.com/react-community/react-native-maps/issues/889 */
jest.mock('react-native-maps', () => {
  const React = require.requireActual('react')
  const MapView = require.requireActual('react-native-maps')

  class MockCallout extends React.Component {
    render() {
      return React.createElement('Callout', this.props, this.props.children)
    }
  }

  class MockMarker extends React.Component {
    render() {
      return React.createElement('Marker', this.props, this.props.children)
    }
  }

  class MockMapView extends React.Component {
    render() {
      return React.createElement('MapView', this.props, this.props.children)
    }
  }

  MockCallout.propTypes = MapView.Callout.propTypes
  MockMarker.propTypes = MapView.Marker.propTypes
  MockMapView.propTypes = MapView.propTypes
  MockMapView.Marker = MockMarker
  MockMapView.Callout = MockCallout
  return MockMapView
})

jest.useFakeTimers()
beforeEach(() => {
  NavigationTestUtils.resetInternalState()
})

it('renders the loading screen', async () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders the root without loading screen', async () => {
  const tree = renderer.create(<App skipLoadingScreen />).toJSON()
  expect(tree).toMatchSnapshot()
})
