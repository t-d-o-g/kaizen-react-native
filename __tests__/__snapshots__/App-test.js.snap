// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the loading screen 1`] = `null`;

exports[`renders the root without loading screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "flex": 1,
            "zIndex": 0,
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column-reverse",
                  "overflow": "hidden",
                },
                Object {
                  "backgroundColor": "#000",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                accessibilityElementsHidden={false}
                collapsable={undefined}
                pointerEvents="auto"
                style={
                  Object {
                    "backgroundColor": "#E9E9EF",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "paddingTop": 0,
                    "position": "absolute",
                    "right": 0,
                    "shadowColor": "black",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0.2,
                    "shadowRadius": 5,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "flex": 1,
                      "height": 1334,
                    }
                  }
                  virtual={undefined}
                >
                  <View
                    onLayout={[Function]}
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#F8F8F8",
                            "borderBottomColor": "#a7a6ab",
                            "borderBottomWidth": 0.5,
                            "elevation": 3,
                            "flexDirection": "row",
                            "height": 64,
                            "justifyContent": "center",
                            "left": 0,
                            "paddingLeft": 6,
                            "paddingRight": 10,
                            "paddingTop": 18,
                            "right": 0,
                            "shadowColor": undefined,
                            "shadowOffset": undefined,
                            "shadowOpacity": undefined,
                            "shadowRadius": undefined,
                            "top": 0,
                          },
                          Object {
                            "backgroundColor": "#282828",
                          },
                        ]
                      }
                      virtual={undefined}
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "flex-start",
                            "alignSelf": "center",
                            "flex": 1,
                          }
                        }
                        virtual={undefined}
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          ellipsizeMode="tail"
                          onPress={[Function]}
                          style={
                            Array [
                              Object {
                                "color": undefined,
                                "fontSize": 12,
                              },
                              Array [
                                Object {
                                  "color": "#000",
                                  "fontSize": 30,
                                },
                                Object {
                                  "color": "#000",
                                  "fontSize": 30,
                                },
                                Object {
                                  "color": "white",
                                },
                              ],
                              Object {
                                "fontFamily": "Ionicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                          virtual={undefined}
                        >
                          ïŒª
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "flex": 1,
                          }
                        }
                        virtual={undefined}
                      />
                      <View
                        style={
                          Object {
                            "alignItems": "flex-end",
                            "alignSelf": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "flex-end",
                          }
                        }
                        virtual={undefined}
                      >
                        <View
                          accessibilityComponentType={undefined}
                          accessibilityLabel={undefined}
                          accessibilityTraits={undefined}
                          accessible={true}
                          collapsable={undefined}
                          hasTVPreferredFocus={undefined}
                          hitSlop={undefined}
                          isTVSelectable={true}
                          nativeID={undefined}
                          onLayout={undefined}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "alignSelf": null,
                              "backgroundColor": "transparent",
                              "borderBottomWidth": null,
                              "borderColor": null,
                              "borderLeftWidth": null,
                              "borderRadius": 50,
                              "borderRightWidth": null,
                              "borderTopWidth": null,
                              "borderWidth": undefined,
                              "elevation": 0,
                              "flexDirection": "row",
                              "height": 45,
                              "justifyContent": "space-between",
                              "marginRight": -9,
                              "opacity": 1,
                              "paddingBottom": 6,
                              "paddingLeft": 15,
                              "paddingRight": 15,
                              "paddingTop": 6,
                              "shadowColor": null,
                              "shadowOffset": null,
                              "shadowOpacity": null,
                              "shadowRadius": null,
                            }
                          }
                          testID={undefined}
                          tvParallaxProperties={undefined}
                        >
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "color": "white",
                              }
                            }
                          >
                            Login
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <MapView
                      initialRegion={
                        Object {
                          "latitude": 40.73136,
                          "latitudeDelta": 0.00922,
                          "longitude": -74.0628,
                          "longitudeDelta": 0.00421,
                        }
                      }
                      onPress={[Function]}
                      region={
                        Object {
                          "latitude": 37.78825,
                          "latitudeDelta": 0.00922,
                          "longitude": -122.4324,
                          "longitudeDelta": 0.00421,
                        }
                      }
                      showsUserLocation={true}
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                      zoomEnabled={true}
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="none"
        style={
          Object {
            "backgroundColor": "#000",
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1000,
          }
        }
        testID={undefined}
      />
      <View
        accessibilityViewIsModal={false}
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "white",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": null,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": -320,
              },
            ],
            "width": 320,
            "zIndex": 1001,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "flex": 1,
                "height": 1334,
              }
            }
            virtual={undefined}
          >
            <View
              onLayout={[Function]}
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#F8F8F8",
                      "borderBottomColor": "#a7a6ab",
                      "borderBottomWidth": 0.5,
                      "elevation": 3,
                      "flexDirection": "row",
                      "height": 64,
                      "justifyContent": "center",
                      "left": 0,
                      "paddingLeft": 6,
                      "paddingRight": 10,
                      "paddingTop": 18,
                      "right": 0,
                      "shadowColor": undefined,
                      "shadowOffset": undefined,
                      "shadowOpacity": undefined,
                      "shadowRadius": undefined,
                      "top": 0,
                    },
                    Object {
                      "backgroundColor": "#606060",
                      "height": 150,
                    },
                  ]
                }
                virtual={undefined}
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "flex": 1,
                    }
                  }
                  virtual={undefined}
                >
                  <Image
                    large={true}
                    source={1}
                    square={true}
                    style={
                      Array [
                        Object {
                          "borderRadius": 0,
                          "height": 80,
                          "width": 80,
                        },
                        Object {
                          "alignSelf": "center",
                        },
                      ]
                    }
                    virtual={undefined}
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.4)",
            "flex": 1,
            "justifyContent": "flex-end",
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#fff",
              "elevation": 4,
              "height": NaN,
              "maxHeight": 667,
              "minHeight": 56,
              "opacity": 1,
              "padding": 15,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        />
      </View>
    </Modal>
  </View>
</View>
`;
